src/librustc_resolve/lib.rs

```diff
@@ -894,7 +895,7 @@ impl<'a> ModuleS<'a> {
 
     fn is_normal(&self) -> bool {
         match self.def {
-            Some(Def::Mod(_)) | Some(Def::ForeignMod(_)) => true,
```

Comment:
`is_normal` means something that `self` or `super` resolve to -- it is used when resolving path prefixes.
`ForeignMod`s should never have been normal, but happened to not make a difference before this PR. Now, `ForeignMod` can't be normal since otherwise a `pub(self)` item in a foreign module would be restricted to the foreign module instead of its containing module.